helm/vitess/templates/_vttablet.tpl

```diff
@@ -377,6 +377,36 @@ spec:
           name: vitess-cm
           key: db.flavor
 
+    lifecycle:
+      preStop:
+        exec:
+          command: ["bash"]
+          args:
+            - "-c"
+            - |
+              set -ex
+
+              VTCTLD_SVC=vtctld.{{ $namespace }}:15999
+
+              master_alias=$(vtctlclient -server $VTCTLD_SVC GetShard {{ $keyspace.name }}/{{ $shard.name }} | jq '.master_alias.uid')
+              current_alias=$(cat /vtdataroot/tabletdata/tablet-uid)
+              if [ $master_alias != $current_alias ]; then
+                  # since this isn't the master, there's no reason to reparent
+                  exit
+              fi
+
+              # reparent before shutting down
+              eval exec /vt/bin/vtctl $(cat <<END_OF_COMMAND
+                -topo_implementation="etcd2"
+                -topo_global_root=/vitess/global
+                -topo_global_server_address="etcd-global-client.{{ $namespace }}:2379"
+                -logtostderr=true
+                -stderrthreshold=0
+                PlannedReparentShard -keyspace_shard={{ $keyspace.name }}/{{ $shard.name }}
```

Comment:
do you want to set `-avoid_master=` here

Reply:
Good point